sn-api.js

```diff
@@ -0,0 +1,31 @@
+
+const request = require('request')
+const btoa = require('btoa')
+const argv = require('yargs').argv
+const urlencode = require('urlencode')
+const baseurl = "https://gpietro3demo.service-now.com"
+auth = 'Basic '+btoa(`${argv.username||process.env.username}`+':'+`${argv.password||process.env.password}`)
+
+var search_string = urlencode(argv.search_string)
+var options = {
+  uri: baseurl + `/api/now/table/kb_knowledge?sysparm_query=123TEXTQUERY321%3D${search_string}&sysparm_limit=3&workflow_state=published`,
```

Comment:
this `123TEXTQUERYSTRING321=` is required to enable the servicenow backend to do an actual search with the query string you send.